src/Symfony/Component/Lock/Store/MongoDbStore.php

```diff
@@ -120,7 +120,7 @@ public function __construct($mongo, array $options = [], float $initialTtl = 300
             if (isset($parsedUrl['query'])) {
                 parse_str($parsedUrl['query'], $query);
             }
-            $this->options['collection'] = $this->options['collection'] ?? $query['collection'] ?? null;
+            $this->options['collection'] = $query['collection'] ?? $this->options['collection'] ?? null;
```

Comment:
Isn't this a BC Break? someone's application could start writing to the wrong database if they are specifying both for some reason and they differ.

Reply:
I would rather we deal with `Lock\Store\MongoDbStore` in https://github.com/symfony/symfony/pull/37218

Reply:
see https://github.com/symfony/symfony/pull/37218#discussion_r439951135 about BC vs bugfix

Reply:
>  there's the path in the uri which should take precedence over $options['database']

@jderusse can you have a look at this, also in the other PR?

Reply:
good catch. Also fix precedenceof credentials in that PR

Reply:
did not found precendence in #37268 @nicolas-grekas . Did you have a specific thing in mind?